src/main/java/duke/helper/Parser.java

```diff
@@ -0,0 +1,144 @@
+package duke.helper;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+import duke.exception.DukeException;
+import duke.exception.EmptyTaskException;
+import duke.exception.InvalidDateTimeException;
+import duke.exception.InvalidTaskException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * Parser class to parse inputs
+ */
+public class Parser {
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructor for Parser class
+     *
+     * @param tasks list of all tasks
+     * @param ui class that handles user interaction
+     */
+    public Parser(TaskList tasks, Ui ui) {
+        this.tasks = tasks;
+        this.ui = ui;
+    }
+
+    /**
+     * Parse user inputs
+     *
+     * @param desc description of the task
+     * @return String representation of Task
+     * @throws DukeException If input is invalid
+     */
```

Comment:
It seems that a lot of logic is crammed into this method.

In the event that this method bugs out due to one of the methods, it can be difficult to find the error point and the offending switch case. 

I would suggest encapsulating the logic for each switch block in its own `Command` class, and all of them may have a `simulate` method that handles their own errors and outputs a UI regardless. This also makes it easy to define test cases for each of the commands and isolate the logic there.
Also, this would in the future make it easier to change the logic for each command, as you would only need to modify the target Command class' `simulate` method.